tests/convergence/rates.cpp

```diff
@@ -0,0 +1,338 @@
+      default: break;
+   }
+   FiniteElementSpace *fespace = new FiniteElementSpace(mesh, fec);
+
+   // 8. Define the solution vector x as a parallel finite element grid function
+   //     corresponding to fespace.
+   GridFunction x(fespace);
+   x = 0.0;
+   // 9. Set up the linear form b(.) and the bilinear form a(.,.).
+   FunctionCoefficient *f=nullptr;
+   FunctionCoefficient *u=nullptr;
+   FunctionCoefficient *divU=nullptr;
+   VectorFunctionCoefficient *U=nullptr;
+   VectorFunctionCoefficient *gradu=nullptr;
+   VectorFunctionCoefficient *curlU=nullptr;
+
+   ConstantCoefficient one(1.0);
+   ConstantCoefficient (1.0);
+   LinearForm b(fespace);
+   BilinearForm a(fespace);
+
+   switch (prob)
+   {
+      case 0:
+         //(grad u_ex, grad v) + (u_ex,v)
+         u = new FunctionCoefficient(u_exact);
+         gradu = new VectorFunctionCoefficient(dim,gradu_exact);
+         b.AddDomainIntegrator(new DomainLFGradIntegrator(*gradu));
+         b.AddDomainIntegrator(new DomainLFIntegrator(*u));
+
+         // (grad u, grad v) + (u,v)
+         a.AddDomainIntegrator(new DiffusionIntegrator(one));
+         a.AddDomainIntegrator(new MassIntegrator(one));
+
+         break;
+      case 1:
+         //(curl u_ex, curl v) + (u_ex,v)
+         U = new VectorFunctionCoefficient(dim,U_exact);
+         curlU = new VectorFunctionCoefficient((dim ==3)?dim:1,curlU_exact);
+         b.AddDomainIntegrator(new VectorFEDomainLFCurlIntegrator(*curlU));
+         b.AddDomainIntegrator(new VectorFEDomainLFIntegrator(*U));
+
+         // (curl u, curl v) + (u,v)
+         a.AddDomainIntegrator(new CurlCurlIntegrator(one));
+         a.AddDomainIntegrator(new VectorFEMassIntegrator(one));
+         break;
+
+      case 2:
+         //(div u_ex, div v) + (u_ex,v)
+         U = new VectorFunctionCoefficient(dim,U_exact);
+         divU = new FunctionCoefficient(divU_exact);
+         b.AddDomainIntegrator(new VectorFEDomainLFDivIntegrator(*divU));
+         b.AddDomainIntegrator(new VectorFEDomainLFIntegrator(*U));
+
+         // (div u, div v) + (u,v)
+         a.AddDomainIntegrator(new DivDivIntegrator(one));
+         a.AddDomainIntegrator(new VectorFEMassIntegrator(one));
+         break;
+
+      case 3:
+         u = new FunctionCoefficient(u_exact);
+         f = new FunctionCoefficient(rhs_func);
+         gradu = new VectorFunctionCoefficient(dim,gradu_exact);
+         b.AddDomainIntegrator(new DomainLFIntegrator(*f));
+         b.AddBdrFaceIntegrator(
+      new DGDirichletLFIntegrator(*u, one, sigma, kappa));
+         a.AddDomainIntegrator(new DiffusionIntegrator(one));
+         a.AddInteriorFaceIntegrator(new DGDiffusionIntegrator(one, sigma, kappa));
+         a.AddBdrFaceIntegrator(new DGDiffusionIntegrator(one, sigma, kappa));
+         break;   
+
+      default:
+         break;
+   }
+   // 10. Perform successive refinements, compute the errors and the
+   //     corresponding rates of convergence
+   Convergence rates;
```

Comment:
For a second I was confused why this compiled. 😄

Reply:
yeah...my naming choices might not be the best...I am open to suggestions.

Reply:
Maybe `ConvergenceStudy`? I really think this is fine though. I'm just not used to seeing stuff in C++ that also parses as normal English!

Reply:
I don't have a strong preference here.

Reply:
Renamed it to ConvergenceStudy.